packages/gitbook-docs/src/markdoc/attributes.ts

```diff
@@ -16,13 +16,23 @@ export class BooleanOrNullAttr {
   };
 }
 
-const convertLink = (link: string): string =>
-  link
+const convertLink = (link: string): string => {
+  const replaced = link
     .replace('/README.md', '')
     .replace('README.md', '')
     .replace('.md', '')
     .replace(new RegExp('^(.*?)\\/?$'), '$1');
 
+  if (replaced.match(/\/#/)) {
+    const splitted = replaced.split('/');
+    const anchor = splitted[splitted.length - 1].replace('/', '');
+    const url = splitted.slice(0, -1).join('/');
+    return `${url}${anchor}`;
+  }
+
+  return replaced;
+};
+
```

Comment:
I think the following implementation is equivalent:
```suggestion
const convertLink = (link: string): string => 
  link
    .replace('/README.md', '')
    .replace('README.md', '')
    .replace('.md', '')
    .replace(new RegExp('^(.*?)\\/?$'), '$1')
    // uniform links like 'this/is/a/path/#fragment' to 'this/is/a/path#fragment'
    .replace(new RegExp('/#'), '#');

```